File: Python/functions/no checking/disarium.py
#Disarium no.
def check(n,c):
    s=0
    while(n>0):
        s+=(n%10)**c
        c-=1
        n//=10
    return s
n=int(input("Enter the no.= "))
c=len(str(n))
if n==check(n,c):
    print("Disarium no.")
else:
    print("Not Disarium No.")

File: C Programms/No. Checking/disarium.c
//WAP to check Disarium no.
#include<stdio.h>
void main()
{
    int n,p,i,s=0,c=0,x;
    printf("Enter the no. = ");
    scanf(" %d",&n);
    for(x=n;n!=0;n/=10)
        c++;
    for(n=x;n!=0;n/=10,c--)
    {
        p=1;
        for(i=1;i<=c;i++)
            p=p*(n%10);
        s+=p;
    }
    (s==x)?printf("%d is an Disarium No.",x):printf("%d is not an Disarium No.",x);
}